handlers/get_company_directors.go

```diff
@@ -1,12 +1,58 @@
 package handlers
 
-import "net/http"
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
 
-// GetCompanyDirectors returns a list of valid company directors that can apply for an auth code
-func GetCompanyDirectors(w http.ResponseWriter, req *http.Request) {
+	"github.com/companieshouse/emergency-auth-code-api/service"
 
-	// TODO :- Add logic to return company directors
+	"github.com/companieshouse/emergency-auth-code-api/models"
 
-	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(http.StatusOK)
+	"github.com/gorilla/mux"
+
+	"github.com/companieshouse/emergency-auth-code-api/utils"
+
+	"github.com/companieshouse/chs.go/log"
+)
+
+// GetCompanyOfficersHandler returns a list of valid company officers that can apply for an auth code
+func GetCompanyOfficersHandler(svc *service.OfficersService) http.Handler {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		vars := mux.Vars(req)
+		companyNumber := vars["company_number"]
+		if companyNumber == "" {
+			log.ErrorR(req, fmt.Errorf("no company number provided in request"))
+			m := models.NewMessageResponse(fmt.Sprintf("no company number provided in request"))
+			utils.WriteJSONWithStatus(w, req, m, http.StatusBadRequest)
+			return
+		}
+
+		// return list of officers from officers service interface
+		companyOfficers, err := svc.GetListOfCompanyOfficers(companyNumber)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("error receiving data from officer database: %v", err))
+			m := models.NewMessageResponse(fmt.Sprintf("error receiving data from officer database: %v", err))
+			utils.WriteJSONWithStatus(w, req, m, http.StatusInternalServerError)
+			return
+		}
+
+		// return 404 if no officers found
+		if companyOfficers.TotalCount == 0 {
+			m := models.NewMessageResponse(fmt.Sprintf("no officer found for company: %s", companyNumber))
+			utils.WriteJSONWithStatus(w, req, m, http.StatusNotFound)
+			return
+		}
+
+		// prepare response to request
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
```

Comment:
Could this be moved to the end of the function, to avoid a failure writing `http.StatusInternalServerError`?